### PR TITLE
Wrong number of pods attempted fix

### PR DESCRIPTION
**Pull request by yaron2**
Taking a shot at #24...

### CODE REVIEW

File: creator.ts [hunk 1]
```diff
@@ -80,6 +84,11 @@ export async function ContainerCreator(client: api.Core_v1Api, startDate: Date,
             }
             pod.spec.containers = containers;
             pod.status.containerStatuses = containerStatuses;
+
+            if (existingPod) {
+                pod.metadata = existingPod.metadata
```
Comment (lachie83) [inline creator.ts]: I'm wondering if we should also set the pod.spec from the existingPod. Thoughts?

File: creator.ts [hunk 2]
```diff
@@ -80,6 +84,11 @@ export async function ContainerCreator(client: api.Core_v1Api, startDate: Date,
...
+            }
+
             try {
                 if (!exists) {
```
Comment (lachie83) [inline creator.ts]: I don't believe that this if statement should be here. I don't think we want to be creating pods in Kubernetes from what's running in ACI. Thoughts? Let me experiment and confirm the behavior if I remove it. In any case, Kubernetes must be the source of truth.

Comment (lachie83) [reply creator.ts]: BTW - I know that it's not part of the PR but it might be worth cleaning up at the same time.

**Comment by neilpeterson** [comment]
Not sure if it will be helpful at this point, but I've detailed my observation on the duplicate pods issue in issue https://github.com/Azure/aci-connector-k8s/issues/24

**Comment by lachie83** [comment]
Hi @neilpeterson. This is indeed helpful. The root of the issue here is that we are currently truncating the labels in the metadata which is required for RC and RS resources to function correctly.

File: creator.ts (hunk 2 above)
Comment (lachie83) [reply creator.ts]: Yeah I think we should remove the the if condition and run the replace. Given that should we rename the existingPod variable to podStatus and yank all the exists code?

**Comment by lachie83** [comment]
cc @brendandburns

**Comment by brendandburns** [comment]
lgtm.

**Pull request merged**